File: elixir/test/merchant_portal_services_api_entity_test.exs
# MerchantPortalServicesApi entity test (offline, mock transport)

defmodule BluefinTecsMerchantServices.MerchantPortalServicesApiEntityTest do
  use ExUnit.Case

  alias Voxgig.Struct, as: S
  alias BluefinTecsMerchantServices.Helpers, as: H
  alias BluefinTecsMerchantServices.Json

  defp fixture do
    Json.parse(File.read!("../.sdk/test/entity/merchant_portal_services_api/MerchantPortalServicesApiTestData.json"))
  end

  defp mk_sdk do
    existing = H.or_(S.getpath(fixture(), "existing"), S.jm([]))
    BluefinTecsMerchantServices.test(S.jm(["entity", existing]))
  end

  defp first_id do
    existing = H.or_(S.getpath(fixture(), "existing.merchant_portal_services_api"), S.jm([]))
    keys = S.keysof(existing)
    if keys == [], do: nil, else: hd(keys)
  end

  test "should create instance" do
    sdk = BluefinTecsMerchantServices.test()
    ent = BluefinTecsMerchantServices.merchant_portal_services_api(sdk)
    assert ent != nil
  end

  test "should create then read back" do
    sdk = BluefinTecsMerchantServices.test(S.jm(["entity", S.jm(["merchant_portal_services_api", S.jm([])])]))
    ent = BluefinTecsMerchantServices.merchant_portal_services_api(sdk)
    created = BluefinTecsMerchantServices.Entity.MerchantPortalServicesApi.create(ent, S.jm(["name", "test-create"]))
    made = BluefinTecsMerchantServices.EntityBase.data_get(created)
    assert S.ismap(made)
    assert S.getprop(made, "id") != nil
  end
end

File: scala/sdktest/MerchantPortalServicesApiEntityTest.scala
// Generated basic-flow test for the merchant_portal_services_api entity (model-driven;
// mirrors the java TestEntity generator). A dependency-free scala-cli test
// object driven by SdkEntityTestMain. Runs against the in-memory test
// transport seeded with the shipped MerchantPortalServicesApiTestData.json fixtures.

import java.util.{ArrayList, LinkedHashMap, List => JList, Map => JMap}

import voxgig.bluefintecsmerchantservicessdk.core.{Helpers, SdkEntity, BluefinTecsMerchantServicesSDK}
import voxgig.bluefintecsmerchantservicessdk.utility.struct.Struct

object MerchantPortalServicesApiEntityTest {

  def run(rep: SdkTestReport): Unit = {
    rep.scope("merchant_portal_services_api.instance") {
      val testsdk = BluefinTecsMerchantServicesSDK.testSDK()
      val ent = testsdk.merchantPortalServicesApi(null)
      rep.check("merchant_portal_services_api.instance", ent != null, "expected non-null merchant_portal_services_api entity")
    }

    rep.scope("merchant_portal_services_api.basic") {
      val entityData = Helpers.toMapAny(SdkTestSupport.readJson(
          "../.sdk/test/entity/merchant_portal_services_api/MerchantPortalServicesApiTestData.json"))
      val options = new LinkedHashMap[String, Object]()
      options.put("entity", entityData.get("existing"))
      val client = BluefinTecsMerchantServicesSDK.testSDK(options, null)

      val idmap = new LinkedHashMap[String, Object]()
      idmap.put("merchant_portal_services_api01", "MERCHANT_PORTAL_SERVICES_API01")
      idmap.put("merchant_portal_services_api02", "MERCHANT_PORTAL_SERVICES_API02")
      idmap.put("merchant_portal_services_api03", "MERCHANT_PORTAL_SERVICES_API03")
      val now = System.currentTimeMillis()

      // CREATE
      val merchantPortalServicesApiRef01Ent = client.merchantPortalServicesApi(null)
      var merchantPortalServicesApiRef01Data = Helpers.toMapAny(Struct.getprop(
          Struct.getpath(entityData, "new.merchant_portal_services_api"), "merchant_portal_services_api_ref01"))
      val merchantPortalServicesApiRef01DataResult = merchantPortalServicesApiRef01Ent.create(merchantPortalServicesApiRef01Data, null)
      merchantPortalServicesApiRef01Data = Helpers.toMapAny(merchantPortalServicesApiRef01DataResult match { case e: SdkEntity => e.data(); case o => o })
      rep.check("merchant_portal_services_api.create.map", merchantPortalServicesApiRef01Data != null, "expected create result to be a map")
    }
  }
}
